lnwallet/chainfee/filtermanager.go

```diff
@@ -0,0 +1,285 @@
+package chainfee
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/rpcclient"
+)
+
+const (
+	// fetchFilterInterval is the interval between successive fetches of
+	// our peers' feefilters.
+	// TODO: Decide a good value for this.
+	//       - The average feefilter broadcast is every 10 minutes, but
+	//         there is also network churn.
+	//       - We want to pick a value frequent enough to keep up with the
+	//         network, but not so frequent that we're pulling the same
+	//         data over and over.
+	fetchFilterInterval = time.Minute * 5
+
+	// medianWindow is the number of historical values that the
+	// filterManager will track.
+	// TODO: Decide a good value for this.
+	//	 - We want a value large enough to smooth out variance, but
+	//         not so large that long-past irrelevant feefilters influence
+	//         us.
+	medianWindow = 25
+)
+
+var (
+	// errNoFilters is an error that's returned if fetchMedianFilter is
+	// called and there is no data available.
+	errNoFilters = fmt.Errorf("no data available")
+)
+
+// filterManager is responsible for determining an acceptable minimum fee to
+// use based on our peers' feefilter values.
+type filterManager struct {
+	// values stores the historical feefilter values we'll use to come up
+	// with an acceptable minimum feerate.
+	values    []SatPerKWeight
+	valuesMtx sync.RWMutex
+
+	fetchFunc func(*rpcclient.Client) ([]SatPerKWeight, error)
```

Comment:
I think we can abstract the fetching further. Assuming there's some closure that holds the client, it just becomes a normal call back. Then we also don't need to worry about bitcoind or not below as well.

Reply:
Another route is making the `fetchFunc` into an interface, then having one version for `btcd` and the other for `bitcoind`. Then when we make the `filterManager` in the respective fee estimators, we pick the one based on the target backend.

Reply:
Made it into a callback that holds the client, lmk what you think